controllers/imagerepository_controller.go

```diff
@@ -432,3 +444,46 @@ func (r *ImageRepositoryReconciler) recordSuspension(ctx context.Context, imageR
 		r.MetricsRecorder.RecordSuspend(*objRef, imageRepo.Spec.Suspend)
 	}
 }
+
+func parseAuthMap(config dockerConfig) (map[string]authn.AuthConfig, error) {
+	auth := map[string]authn.AuthConfig{}
+	for host, entry := range config.Auths {
+		u, err := parseURL(host)
+		if err != nil {
+			return nil, err
+		}
+
+		host = u.Host
+
+		auth[host] = entry
+	}
+
+	return auth, nil
+}
+
+func parseURL(urlStr string) (*url.URL, error) {
+	if urlStr == "http://" || urlStr == "https://" {
+		return nil, errors.New("Empty registry auth url")
+	}
+
+	// Some users were passing in credentials in the form of
+	// http://docker.io and http://docker.io/v1/, etc.
+	// So strip everything down to the host.
+	// Also, the registry might be local and on a different port.
+	// So we need to check for that because url.Parse won't parse the ip:port format very well.
+	u, err := url.Parse(urlStr)
+
+	// if anything went wrong try to prepend https://
+	if err != nil || u.Host == "" {
+		u, err = url.Parse(fmt.Sprintf("https://%s/", urlStr))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if u.Host == "" {
+		return nil, errors.New(fmt.Sprintf("Invalid registry auth url: %s. Must be a valid http address (e.g. https://gcr.io/v1/)", urlStr))
+	}
+
+	return u, nil
```

Comment:
Considering that in all use cases only the `u.Host` attribute ends up being used, it might make sense to just return that and maybe rename the func to `getURLHost`.